### ISSUE TITLE
Allow automatic CCM/CSI migration

### ISSUE THREAD

**Issue by xmudrii** [kubermatic/kubermatic#6267]

**Comment by mlavacca** [comment]
@kron4eg @xmudrii @irozzo-1A Reading at the epic history I got that the way you thought to perform the automatic migration is through a mutating webhook that detects the need for migration and performs the following changes:
* adds the `externalCloudProvider` feature to the cluster
* adds the following annotations to the cluster
  * `csi-migration.k8c.io/migration-needed`
  * `ccm-migration.k8c.io/migration-needed`

Is it the issue's scope?

EDIT: Actually I saw that [this](https://github.com/kubermatic/kubermatic/issues/6555) issue already implements this mechanism, therefore my thought was wrong. At this point, it is not clear to me the scope of this issue. Any hint?
